File: typescript/Behavioral/18-Observer/Observer/SupervisorAlert.ts
import type { JobCard } from "../Subject/JobCard";
import { IJobCardObserver } from "./IJobCardObserver";

export class SupervisorAlert implements IJobCardObserver {
  constructor(private readonly supervisorName: string) {}

  update(eventName: string, jobCard: JobCard): void {
    console.log(
      `  🔔 [${this.supervisorName}] Alert: '${eventName}' — Assignee: ${jobCard.assignee ?? "unassigned"}`,
    );
  }
}

File: csharp/Behavioral/18-Observer/Observer/SupervisorAlert.cs
namespace Observer;

public class SupervisorAlert(string supervisorName) : IJobCardObserver
{
    public void Update(string eventName, JobCard jobCard)
    {
        Console.WriteLine(
            $"  🔔 [{supervisorName}] Alert: '{eventName}' — Assignee: {jobCard.Assignee ?? "unassigned"}"
        );
    }
}
